src/components/slider/slider.ts

```diff
@@ -215,6 +228,68 @@ export class MdSlider implements AfterContentInit {
   }
 
   /**
+   * Calculates the separation in pixels of tick marks. If there is no tick interval or the interval
+   * is set to something other than a number or 'auto', nothing happens.
+   */
+  private _updateTickSeparation() {
+    if (this._tickInterval == 'auto') {
+      this._updateAutoTickSeparation();
+    } else if (Number(this._tickInterval)) {
+      this._updateTickSeparationFromInterval();
+    }
+  }
+
+  /**
+   * Calculates the optimal separation in pixels of tick marks based on the minimum auto tick
+   * separation constant.
+   */
+  private _updateAutoTickSeparation() {
+    let width = this._sliderDimensions.width;
+
+    // This calculates which step is far enough away from the beginning of the slider to draw a tick
+    // at. This value will then be used to draw ticks at every tickStep steps.
+    let tickStep =
```

Comment:
I think this would benefit from some more explanation as to where the calculation came from.